### PR TITLE
sys/shell/vfs: add umount, rmount and mkdir commands

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Follow up to #18109, add the shell commands.

### Testing procedure

Run the commands.

### CODE REVIEW

File: sys/vfs/vfs.c [hunk 1]
```diff
@@ -1149,6 +1149,22 @@ static int _auto_mount(vfs_mount_t *mountp, unsigned i)
...
+    if (res == 0) {
+        return 0;
+    }
+
+    if (res) {
+        DEBUG("vfs%u umount: error %d\n", i, res);
+    }
+
+    return res;
+}
```
Comment (benpicco) [inline sys/vfs/vfs.c:1166]: This is now no longer needed

File: sys/shell/commands/sc_vfs.c [hunk 2]
```diff
@@ -629,6 +688,12 @@ int _vfs_handler(int argc, char **argv)
     else if (MOUNTPOINTS_NUMOF > 0 && strcmp(argv[1], "mount") == 0) {
         return _mount_handler(argc - 1, &argv[1]);
     }
+    else if (MOUNTPOINTS_NUMOF > 0 && strcmp(argv[1], "umount") == 0) {
+        return _umount_handler(argc - 1, &argv[1]);
+    }
+    else if (MOUNTPOINTS_NUMOF > 0 && strcmp(argv[1], "rmount") == 0) {
```
Comment (benpicco) [inline sys/shell/commands/sc_vfs.c:694]:
```suggestion
    else if (MOUNTPOINTS_NUMOF > 0 && strcmp(argv[1], "remount") == 0) {
```

File: sys/shell/commands/sc_vfs.c [hunk 3]
```diff
@@ -172,6 +180,35 @@ static int _mount_handler(int argc, char **argv)
...
+static int _umount_handler(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s [path]\n", argv[0]);
+        puts("umount pre-configured mount point");
+        return -1;
+    }
+
+    int res = vfs_unmount_by_path(argv[1]);
+    puts(strerror(res));
```
Comment (benpicco) [inline sys/shell/commands/sc_vfs.c:192]: Is `_errno_string` less expensive than `strerror`?
I know I used it for `_mount_handler` but that was probably a mistake given how much ROM it adds.

File: sys/shell/commands/sc_vfs.c [hunk 4]
```diff
@@ -167,8 +175,41 @@ static int _mount_handler(int argc, char **argv)
...
+{
+    if (argc < 2) {
+        printf("usage: %s [path]\n", argv[0]);
+        puts("remount pre-configured mount point");
+        return -1;
+    }
+
+    uint8_t buf[16];
+    int res = vfs_unmount_by_path(argv[1]);
+    if (res == 0) {
```
Comment (benpicco) [inline sys/shell/commands/sc_vfs.c:209]: Shouldn't we try to mount it anyway?

**Comment by benpicco** [comment]
Please squash

**Comment by fjmolinas** [comment]
Squashed!

**Pull request merged**